File: backend/src/main/java/com/pik/backend/services/TeamService.java
package com.pik.backend.services;

import com.pik.ride2work.tables.pojos.Team;

import java.util.concurrent.Future;

import com.pik.ride2work.tables.pojos.User;

import java.util.List;

public interface TeamService {

    Future<User> getTeamOwner(Integer teamId);

    public Future<Void> delete(Integer teamId);

    public Future<Team> create(Team team, Integer ownerId);

    public Future<Team> update(Team team);

    public Future<List<Team>> list();

    public Future<Team> getByName(String name);

    Future<Team> getById(Integer id);

    Future<List<User>> getUserList(Integer teamId);
}

File: frontend/src/main/web/src/app/_services/team.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Team} from "../_models/team";
import {User} from "../_models/user";

@Injectable({
  providedIn: 'root'
})
export class TeamService {
  constructor(private http: HttpClient) { }

  create(team: Team, ownerId: number) {
    return this.http.post<any>(`/api/teams/${ownerId}`, team);
  }

  update(team:Team) {
    return this.http.put<any>(`/api/teams`, team);
  }

  list() {
    return this.http.get<any>(`/api/teams/all`);
  }

  getByName(name: string){
    return this.http.get<Team>(`/api/teams/byName/${name}`);
  }

  getById(id: number) {
    return this.http.get<Team>(`/api/teams/${id}`);
  }

  delete(id: number) {
    return this.http.delete(`/api/teams/${id}`);
  }

  getOwner(id: number) {
    return this.http.get<User>(`/api/teams/owner/${id}`);
  }

  listUsers(id: number) {
    return this.http.get<User[]>(`/api/teams/listUsers/${id}`);
  }
}
